stix_shifter_modules/tanium/stix_transmission/connector.py

```diff
@@ -0,0 +1,133 @@
+
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.stix_transmission.utils.RestApiClientAsync import RestApiClientAsync
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+
+
+class Connector(BaseJsonSyncConnector):
+    _QUERY_ENDPOINT = 'plugin/products/threat-response/api/v1/alerts?'
+    _QUERY_OFFSET = '&offset='
+    
+    def __init__(self, connection, configuration):
+        self.config = TaniumConfig(connection, configuration)
+        self.headers = {'session': self.config.getAccessToken(),
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'}
+        self.api_client = RestApiClientAsync(self.config.getHostname(), port=self.config.getPort(), headers=self.headers)
+        self.final_results = list()
+        self.current_offset = 0
+        self.return_obj = dict()
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+    
+    async def ping_connection(self):
+        self.return_obj['success'] = False
+        try:
+            await self.query_tanium_api(self._QUERY_ENDPOINT)
+            self.return_obj['success'] = True
+        except Exception:
+            self.logger.error('error when pinging Tanium datasource {}:'.format(self.return_obj["error"]))
+        return self.return_obj
+        
+
+    async def create_results_connection(self, query, offset, limit):
+        self.return_obj['success'] = False
+        self.return_obj["data"] = {}
+        offset = int(offset)
+        limit = int(limit) + 1
+        #Initalize values
+        self.current_offset = offset
+        #This can be any value up to 500.
+        max_per_query_limit = 500
+        
+        if(limit < max_per_query_limit):
+            per_query_limit = limit
+        
+        try:
+            results = await self.get_results(per_query_limit, query, self.current_offset) 
+            #Are we done?
+            while(len(self.final_results) < limit and len(results) > 0):
+                results = await self.get_results(per_query_limit, query, self.current_offset)
+                
+            self.return_obj["data"] = self.final_results
+            self.return_obj['success'] = True
+            self.return_obj['metadata'] = self.return_obj['metadata'] = {"next_offset": self.current_offset, "total_result_count": len(self.final_results)}
+            
+        except Exception as err:
+            self.logger.error('error when connecting to the Tanium datasource {}:'.format(self.return_obj["error"]))
+        return self.return_obj
+        
+    async def get_results(self, per_query_limit, query, current_offset):
+        #Create initial query
+        if(query != ""):
+            current_query = "%s%s&limit=%s&offset=%s&expand=intelDoc" % (self._QUERY_ENDPOINT, query, per_query_limit, current_offset)
```

Comment:
string formatting using %s is old and slow. python has better option called f-string https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-pep498

You would probably see .format() method used as well in different modules. But i think we should use f-string from now on.